ref/cmake/compiler_flags_Intel_Fortran.cmake

```diff
@@ -2,19 +2,19 @@
 # COMMON FLAGS
 ####################################################################
 
-set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -traceback -i4 -r8" )
+set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -traceback -i4 -r8 -heap-arrays -fp-model precise " )
 
 ####################################################################
 # RELEASE FLAGS
 ####################################################################
 
-set( CMAKE_Fortran_FLAGS_RELEASE "-O3 -xHost -qopt-zmm-usage=high " )
+set( CMAKE_Fortran_FLAGS_RELEASE "-O3 -xHost " )
 
 ####################################################################
 # DEBUG FLAGS
 ####################################################################
 
-set( CMAKE_Fortran_FLAGS_DEBUG   "-g -O0 -debug -nolib-inline -fno-inline-functions -assume protect_parens,minus0 -prec-div -prec-sqrt -check bounds -check uninit -fp-stack-check -init=snan,array -warn unused" )
+set( CMAKE_Fortran_FLAGS_DEBUG   "-O0 -debug -gen-interfaces -warn interfaces -check -fpe0 -nolib-inline -fno-inline-functions -assume protect_parens,minus0 -prec-div -prec-sqrt -init=snan,array -warn unused" )
```

Comment:
Using `-check` instead of `-check bounds -check uninit -fp-stack-check` is probably what caused the 1000s of warning messages in debug.  The `check all` will issue "temporary array" warnings all over the place which we don't care about.  You might consider reverting the code to the original (to preserve the original source) and restore those options and see if you still get the warnings.  If that works, I'd prefer **NOT** to modify the numerical code.

Reply:
Yes, the warning messages were caused by the "-check" flag.  I restored the previous Intel Fortran debug flags and restored the original d2a2c_vect subroutine.